### PR TITLE
fix(components): [select] value support object

### PR DESCRIPTION
**Pull request by btea**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Review by jw-foss — COMMENTED** [review]
@btea Left a comment for you 🏓

File: packages/components/select/src/select.vue [hunk 1]
```diff
@@ -236,7 +236,7 @@
             :class="[
               nsSelect.is(
                 'empty',
-                !allowCreate && query && filteredOptionsCount === 0
+                !allowCreate && !!query && filteredOptionsCount === 0
```
[the remark this answers is not in the document]
Comment (btea) [reply packages/components/select/src/select.vue]: Done 🏓

**Review by jw-foss — APPROVED** [review]

**Comment by cablin** [comment]
I  use object as el-option value, and it works correct. But I updated element-plus from 2.1.4 to 2.1.8 this morning, and find that all the select can not display the option name 
image

**Comment by btea** [comment]
@cablin Can you provide an online link?

**Comment by exherb** [comment]
https://github.com/element-plus/element-plus/pull/7054

**Pull request merged**